src/lib/storage/proposal/devicegraph_generator.rb

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "storage/proposal/refined_devicegraph"
+require "storage/proposal/space_maker"
+require "storage/proposal/partition_creator"
+
+module Yast
+  module Storage
+    module Proposal
+      using RefinedDevicegraph
+
+      # Class to create devicegraphs that can accommodate a given collection of
+      # volumes
+      class DevicegraphGenerator
```

Comment:
The old proposal class is not gone. Is the new `ProposalDemoClient`. What I disliked about the old `Proposal` class is that it did quite some things: deciding the volumes, creating the target devicegraph, calculating the actions for that devicegraph, printing the result and all the intermediate steps if needed and even some additional tasks....

I just wanted to break that block into pieces: deciding the volumes goes now into a class that actually returns the list of volumes, you use those volumes to call a different class that calculates the final map, then you can add that map for the actions... and so on.

The old proposal class is still there. It's simply that at this point in time, it looks like a YaST client to me.